nsq/src/test/scala/com.snowplowanalytics.snowplow.collectors.scalastream/NsqConfigSpec.scala

```diff
@@ -18,8 +18,108 @@
   */
 package com.snowplowanalytics.snowplow.collectors.scalastream
 
-import com.snowplowanalytics.snowplow.collectors.scalastream.config.ConfigSpec
+import cats.effect.testing.specs2.CatsEffect
+import cats.effect.{ExitCode, IO}
+import com.snowplowanalytics.snowplow.collector.core.{Config, ConfigParser}
+import com.snowplowanalytics.snowplow.collectors.scalastream.sinks.NsqSinkConfig
+import org.http4s.SameSite
+import org.specs2.mutable.Specification
 
-class NsqConfigSpec extends ConfigSpec {
-  makeConfigTest("nsq", "", "")
+import java.nio.file.Paths
+import scala.concurrent.duration.DurationInt
+
+class NsqConfigSpec extends Specification with CatsEffect {
+
+  "Config parser" should {
+    "be able to parse extended nsq config" in {
+      assert(
+        resource       = "/config.nsq.extended.hocon",
+        expectedResult = Right(NsqConfigSpec.expectedConfig)
+      )
+    }
+    "be able to parse minimal nsq config" in {
+      assert(
+        resource       = "/config.nsq.minimal.hocon",
+        expectedResult = Right(NsqConfigSpec.expectedConfig)
+      )
+    }
+  }
+
+  private def assert(resource: String, expectedResult: Either[ExitCode, Config[NsqSinkConfig]]) = {
+    val path = Paths.get(getClass.getResource(resource).toURI)
+    ConfigParser.fromPath[IO, NsqSinkConfig](Some(path)).value.map { result =>
+      result must beEqualTo(expectedResult)
+    }
+  }
+}
+
+object NsqConfigSpec {
+  private val expectedConfig = Config[NsqSinkConfig](
+    interface = "0.0.0.0",
+    port      = 8080,
+    paths     = Map.empty[String, String],
+    p3p = Config.P3P(
+      policyRef = "/w3c/p3p.xml",
+      CP        = "NOI DSP COR NID PSA OUR IND COM NAV STA"
+    ),
+    crossDomain = Config.CrossDomain(
+      enabled = false,
+      domains = List("*"),
+      secure  = true
+    ),
+    cookie = Config.Cookie(
+      enabled        = true,
+      expiration     = 365.days,
+      name           = "sp",
+      domains        = List.empty,
+      fallbackDomain = None,
+      secure         = true,
+      httpOnly       = true,
+      sameSite       = Some(SameSite.None)
+    ),
+    doNotTrackCookie = Config.DoNotTrackCookie(
+      enabled = false,
+      name    = "",
+      value   = ""
+    ),
+    cookieBounce = Config.CookieBounce(
+      enabled                 = false,
+      name                    = "n3pc",
+      fallbackNetworkUserId   = "00000000-0000-4000-A000-000000000000",
+      forwardedProtocolHeader = None
+    ),
+    redirectMacro = Config.RedirectMacro(
+      enabled     = false,
+      placeholder = None
+    ),
```

Comment:
Instead of redefining everything you can probably reuse [what is in `TestUtils`](https://github.com/snowplow/stream-collector/blob/feature/http4s/http4s/src/test/scala/com.snowplowanalytics.snowplow.collector.core/TestUtils.scala#L27) and copy it : `TestUtils.testConfig.copy()` (e.g. [here](https://github.com/snowplow/stream-collector/blob/feature/http4s/http4s/src/test/scala/com.snowplowanalytics.snowplow.collector.core/ServiceSpec.scala#L372-L374))

Reply:
I tried, but couldn't get the type conversion to work initially - we don't do this for any of the other config tests, so in the interest of getting it merged I'll leave as is if that's ok.